osu.Game.Rulesets.Osu/Skinning/Legacy/LegacySliderBall.cs

```diff
@@ -94,6 +107,33 @@ protected override void UpdateAfterChildren()
             layerSpec.Rotation = -appliedRotation;
         }
 
+        private void onHitObjectApplied(DrawableHitObject? drawableObject = null)
+        {
+            if (drawableObject != null && drawableObject is not DrawableSlider)
+                return;
```

Comment:
This feels like a very weird conditional. When can it ever not b a `DrawableSlider`? Also, this is blocking the addition of any frames below (on the second null check which I'm equally unsure about).

Reply:
I thought `drawableObject` could be a nested object of `DrawableSlider`, seeing that it's provided as an argument of the event. If that's not the case, we might as well just remove the argument to avoid this confusion.

Reply:
...no? that would never make any sense in any universe.

> If that's not the case, we might as well just remove the argument to avoid this confusion.

i'm not sure a single person has had a problem with this until now. the DHO being in the event is for convenience. are you suggesting that other similar events like `DefaultsApplied` should also not have the same type of param due to "confusion"?

Reply:
> I thought drawableObject could be a nested object of DrawableSlider

You're likely confusing the other method, `ApplyCustomUpdateState` which *does* do this.